ast/compile.go

```diff
@@ -606,6 +606,14 @@ func (c *Compiler) RuleIndex(path Ref) RuleIndex {
 	return r.(RuleIndex)
 }
 
+// PassesTypeCheck determines whether the given body passes type checking
+func (c *Compiler) PassesTypeCheck(body Body) bool {
+	checker := newTypeChecker().WithVarRewriter(rewriteVarsInRef(c.RewrittenVars))
```

Comment:
Since the errors are discarded, we don't have to set the var rewriter on the type checker. The var rewriter just maps rewritten vars back to the user's own (e.g., `x := 1` gets rewritten to something like `__local0__ := 1`  but if an error had to emitted because of how `__local0__` was used, it should refer to `x` not `__local0__` which users have no concept of.)

Reply:
Fixed